src/jobs/NotificationTrigger/index.js

```diff
@@ -19,50 +19,85 @@ module.exports = async function(context) {
 					currentTime,
 				}
 			})
-			if (userWithCurrentTime) {
-				const latestArticle = await newsDbService.getLatestNewsArticle()
-				const todaysTimeFrame = getStartEndTime()
 
-				const todaysNotifications = await NotificationDbService.getNotifications({
-					createdAt: { $gte: todaysTimeFrame.startTime, $lt: todaysTimeFrame.endTime },
-				})
-				if (latestArticle) {
-					const notifications = []
-					for (const user of userWithCurrentTime) {
-						const isSent = todaysNotifications.find(
-							notification =>
-								String(notification.user) === String(user._id) && String(notification.article) === String(latestArticle[0]._id),
-						)
-						if (isSent) {
-							continue
+			const article = await newsDbService.getLatestNewsArticle()

```

Comment:
@chetannn It might be a good idea to break out some functionalities into a separate class making it more readable